docs/content/Examples-Tutorials-Recipes/Joining-multiple-data-sources.md

```diff
@@ -0,0 +1,123 @@
+---
+title: Joining Data from Multiple Data Sources
+permalink: /recipes/joining-multiple-data-sources
+category: Examples & Tutorials
+subCategory: Query acceleration
+menuOrder: 6
+---
+
+## Use case
+
+Let's imagine we store information about products and their suppliers in
+separate databases. We want to aggregate data from multiple data sources and get
+data at lightning speed. In the recipe below, we'll learn how to create the
+[rollup join](https://cube.dev/docs/schema/reference/pre-aggregations#parameters-type-rollupjoin)
+between two databases to achieve our goal.
+
+## Configuration
+
+First of all, we should define our database connections depending on the
+`dataSource` parameter:
+
+<GitHubCodeBlock
+  href="https://github.com/cube-js/cube.js/blob/recipes/cross-datasource-join/examples/recipes/joining-multiple-datasources-data/cube.js"
+  titleSuffixCount={2}
+  part=""
+  lang="js"
+/>
+
+## Data schema
+
+To speed up our queries, we'll use the
+[rollup](https://cube.dev/docs/schema/reference/pre-aggregations#parameters-type-rollup)
+pre-aggregations. Let's define it for products and suppliers.
+
+<GitHubCodeBlock
+  href="https://github.com/cube-js/cube.js/blob/recipes/cross-datasource-join/examples/recipes/joining-multiple-datasources-data/schema/Products.js"
+  titleSuffixCount={2}
+  part="productsRollup"
+  lang="js"
+/>
+
+<GitHubCodeBlock
+  href="https://github.com/cube-js/cube.js/blob/recipes/cross-datasource-join/examples/recipes/joining-multiple-datasources-data/schema/Suppliers.js"
+  titleSuffixCount={2}
+  part="suppliersRollup"
+  lang="js"
+/>
+
+What if we want to get the data from the defined above pre-aggregations in one
+query? We can use the `rollupJoin` pre-aggregation. It will be able to aggregate
+data from various data sources. Let's define it! Notice that the rollup that
+joined should contain dimensions on which they're joined. In our case, is the
+`CUBE.supplierId` dimension in the `Products` cube:
+
+<GitHubCodeBlock
+  href="https://github.com/cube-js/cube.js/blob/recipes/cross-datasource-join/examples/recipes/joining-multiple-datasources-data/schema/Products.js"
+  titleSuffixCount={2}
+  part="combinedRollup"
+  lang="js"
+/>
+
+## Query
+
+Let's get the product names and their suppliers' info, such as company name and
+email, with the following query:
+
+```javascript
+{
+  "order": {
+    "Products.name": "asc"
+  },
+  "dimensions": [
+    "Products.name",
+    "Suppliers.company",
+    "Suppliers.email"
+  ],
+  "limit": 3
+}
+```
+
+## Result
+
+As a result, we'll get the data from two pre-aggregations joined into one
```

Comment:
```suggestion
We'll get the data from two pre-aggregations joined into one
```

Reply:
I think this simplifies the sentence since we already have the `Results` heading above.